types/src/transaction/change_set.rs

```diff
@@ -11,8 +13,35 @@ pub struct ChangeSet {
 }
 
 impl ChangeSet {
-    pub fn new(write_set: WriteSet, events: Vec<ContractEvent>) -> Self {
-        Self { write_set, events }
+    pub fn new(write_set: WriteSet, events: Vec<ContractEvent>) -> Result<Self, VMStatus> {
```

Comment:
n00b Q: @vgao1996 @wrwg  shall I return a `PartialVmError` to be able to use `with_message()`?

Reply:
Your choice. As you now have a descriptive error code, it is fine as is.